### PR TITLE
New AS Instance module

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
Add new module that helps to create/modify/delete AS Instances in AS Group.

### CODE REVIEW

File: plugins/modules/as_group.py [hunk 1]
```diff
@@ -485,11 +485,11 @@ def _attrs_security_groups(self, attrs, security_groups, as_config=None):
             if len(security_groups) == 1:
                 sec_groups = []
                 sec_group = {}
-                group = self.conn.network.find_security_group(
-                    name_or_id=security_groups.id
+                sg_group = self.conn.network.find_security_group(
```
Comment (outcatcher) [inline plugins/modules/as_group.py:488]: `sg_group` is basically `security group group` :)

File: plugins/modules/as_group.py [hunk 2]
```diff
@@ -738,8 +738,8 @@ def _wait_for_instances(self, as_group, timeout, desire_instance_number=0):
                 group=as_group
             ))
             if (len(instances) == desire_instance_number
-                    and [instance.id for instance in instances
-                         if instance.id]):
+                    and len([instance.id for instance in instances if
+                             instance.id]) == desire_instance_number):
```
Comment (outcatcher) [inline plugins/modules/as_group.py:742]: Can be split into
```python
instances_with_id = [instance.id for instance in instances if instance.id]
...
if len(instances) == desire_instance_number == len(instances_with_id):
...
```

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: plugins/modules/as_instance.py [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+    )
+    module_kwargs = dict(
+        supports_check_mode=True
+    )
+
+    def _system_state_change(self, instances, as_instances, state, action):
+        if state == 'present':
+            if action is None:
+                return False
+            elif action == 'remove':
```
Comment (anton-sidelnikov) [inline plugins/modules/as_instance.py:135]: it never happens

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

File: plugins/modules/as_instance.py (hunk 3 above)
Comment (irina-pereiaslavskaia) [reply plugins/modules/as_instance.py:135]: It happens when we will work in check_mode.

**Review by Polina-Gubina — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**